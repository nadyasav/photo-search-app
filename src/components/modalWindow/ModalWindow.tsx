import React from 'react';
import styles from './ModalWindow.module.css';
import styled from 'styled-components';

type ModalWindowAnimType = 'animScale';
interface IModalWindowProps {
  active: boolean;
  activeAnimation?: ModalWindowAnimType;
  setWindowState: (value: boolean) => void;
  children: React.ReactNode;
  styled?: React.ElementType | keyof JSX.IntrinsicElements;
}

function ModalWindow(props: IModalWindowProps) {
  const Window = props.styled || styled.div``;

  function handleWindowClick(e: React.MouseEvent<HTMLDivElement | HTMLButtonElement, MouseEvent>) {
    e.stopPropagation();
  }

  function closeWindow() {
    props.setWindowState(false);
  }

  return (
    <div
      data-testid="modalWindow"
      className={`${styles.window__overlay} ${props.active ? styles.active : ''} ${
        props.activeAnimation ? styles[props.activeAnimation] : ''
      }`}
      onClick={closeWindow}
    >
      <div className={styles.window__outer}>
        <div className={styles.window__box}>
          <Window
            data-testid="modalWindowChild"
            className={`${styles.window}`}
            onClick={handleWindowClick}
          >
            {props.children}
            <button
              data-testid="modalWindowCloseBtn"
              type="button"
              className={styles.close_btn}
              onClick={closeWindow}
            ></button>
          </Window>
        </div>
      </div>
    </div>
  );
}

export default ModalWindow;
