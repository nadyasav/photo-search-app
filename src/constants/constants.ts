export const ROUTES = {
  main: '/',
  about: '/about',
  notFound: '*',
  form: '/form',
  cards: '/cards',
};
export const DEFAULT = 'default';
export const VALIDATION = 'validation';
export const RESET = 'reset';
export const ERROR_TEXT = {
  empty: 'This field cannot be empty.',
  symbols: 'Field must contain only letters, apostrophes and hyphens.',
  date: 'Enter your Date Of Birth.',
  dateRange: 'Date must be between ',
  country: 'Select your country.',
  fileNotSelected: 'No picture selected',
  file: 'Upload a profile picture.',
  switcher: 'Select your gender.',
  agree: 'You must agree.',
};
export const FLICKR_SEARCH_URL =
  'https://www.flickr.com/services/rest/?method=flickr.photos.search';
export const FLICKR_SEARCH_PARAMS =
  'format=json&nojsoncallback=1&media=photos&extras=owner_name,license,views,description,date_taken,url_z,url_l,original_format';
export const FLICKR_SERVER_PHOTOS_URL = 'https://live.staticflickr.com';
export const FLICKR_API_KEY = 'api_key=d2af878f1932e92147f159ded98b5234';
export const FLICKR_SORT = {
  relevance: 'relevance',
  interestingnessAsc: 'interestingness-asc',
  interestingnessDesc: 'interestingness-desc',
  datePostedAsc: 'date-posted-asc',
  datePostedDesc: 'date-posted-desc',
  dateTakenAsc: 'date-taken-asc',
  dateTakenDesc: 'date-taken-desc',
};
export const FLICKR_PHOTO_DEFAULT_FORMAT = 'png';
export const ERROR_FETCH_TEXT = 'Something went wrong! Error message:';
export const ERROR_SEARCH_TEXT = 'No results for';
export const MODALWINDOW_ANIMTYPE = {
  animScale: 'animScale' as const,
};
export const CARDSBOX_MIN_HEIGHT = 200;
