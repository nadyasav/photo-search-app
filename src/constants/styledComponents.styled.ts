import styled from 'styled-components';

export const formSubmitedWindow = styled.div`
  max-width: 380px;
  padding: 40px 20px;
`;

export const cardPopupWindow = styled.div`
  max-width: 1064px;
  width: 100%;
`;

export const preloaderCircle = 'border-color: #ffffff;';
