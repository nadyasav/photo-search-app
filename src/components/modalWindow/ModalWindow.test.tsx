import { render, screen } from '@testing-library/react';
import { MODALWINDOW_ANIMTYPE } from 'constants/constants';
import React from 'react';
import styled from 'styled-components';
import ModalWindow from './ModalWindow';

const setWindowState = jest.fn();

const styledWindow = styled.div`
  width: 100%;
`;

const props = {
  active: true,
  activeAnimation: MODALWINDOW_ANIMTYPE.animScale,
  setWindowState: setWindowState,
  styled: styledWindow,
};

describe('Testing ModalWindow ', () => {
  it('Renders ModalWindow', () => {
    const modalWindow = render(
      <ModalWindow {...props}>
        <div>children</div>
      </ModalWindow>
    );
    expect(modalWindow).toMatchSnapshot();
    expect(screen.getByText(/children/i)).toBeInTheDocument();
  });
});
