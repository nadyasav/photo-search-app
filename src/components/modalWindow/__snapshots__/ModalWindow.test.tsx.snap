// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ModalWindow  Renders ModalWindow 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="window__overlay active animScale"
        data-testid="modalWindow"
      >
        <div
          class="window__outer"
        >
          <div
            class="window__box"
          >
            <div
              class="sc-bcXHqe hngVog window"
              data-testid="modalWindowChild"
            >
              <div>
                children
              </div>
              <button
                class="close_btn"
                data-testid="modalWindowCloseBtn"
                type="button"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="window__overlay active animScale"
      data-testid="modalWindow"
    >
      <div
        class="window__outer"
      >
        <div
          class="window__box"
        >
          <div
            class="sc-bcXHqe hngVog window"
            data-testid="modalWindowChild"
          >
            <div>
              children
            </div>
            <button
              class="close_btn"
              data-testid="modalWindowCloseBtn"
              type="button"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
